lib/addons/backstage/index.ts

```diff
@@ -0,0 +1,112 @@
+import { Construct } from "constructs";
+import { HelmAddOn, HelmAddOnUserProps } from "../helm-addon";
+import { dependable, setPath } from "../../utils";
+import { ClusterInfo, Values } from "../../spi";
+import { ICertificate } from "aws-cdk-lib/aws-certificatemanager";
+import * as rds from "aws-cdk-lib/aws-rds";
+import { CfnOutput } from 'aws-cdk-lib';
+
+const HTTPS = "https://";
+
+/**
+ * User provided options for the Helm Chart
+ */
+export interface BackstageAddOnProps extends HelmAddOnUserProps {
+    subdomain: string,
+    certificateResourceName: string,
+    imageRegistry: string,
+    imageRepository: string,
+    imageTag?: string,
+    databaseResourceName: string
+    databaseSecretTargetName: string
+}
+
+/**
+ * Default props to be used when creating the Helm chart
+ */
+const defaultProps = {
+  name: "blueprints-backstage-addon",
+  namespace: "backstage",
+  chart: "backstage",
+  version: "0.17.0",
+  release: "backstage",
+  repository:  "https://backstage.github.io/charts",
+  imageTag: "latest",
```

Comment:
removed

Reply:
@shapirov103 actually, thinking further about this:
- for context: the image this tag refers to, is the image produced by the user an pushed to ECR
- a tag is always present in Docker images, if not specified it is "latest"
I would like to keep the line 34 above ( imageTag: "latest") because it is part of a default setting (defaultProps), overridden by the user value, if any. if no value provided, removing the default will anyway force the system to pick a tag, which as of today's setting in Docker pull is "latest"; keeping the line will just make explicit a behavior existing  in Docker anyway... so it is more clear from my perspective and causes no change if the user actually selects a tag. Just removing the line will not help fixing the concern of inconsistency you raised: to do that, the user needs to select a tag (tag that is defined outside the addon so we dont know it). I would revert the change I made to remove the line if ok. please let me know

Reply:
Please see my comment against the props object. It is not obvious what this imageTag was referring to, so I think we need to do a better job documenting it. If it is customer image tag, I suppose the current behavior  is acceptable.